graphql_compiler/schema_merging/rename_schema.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import namedtuple
+
+from graphql import build_ast_schema, parse
+from graphql.language.visitor import Visitor, visit
+
+from .utils import SchemaError, get_schema_data
+
+
+RenamedSchema = namedtuple(
+    'RenamedSchema', [
+        'schema_ast',  # type: Document, ast representing the renamed schema
+        'reverse_name_map',  # type: Dict[str, str], new type name to old type name
+        'reverse_root_field_map'  # type: Dict[str, str], new root field name to old root field
+    ]
+)
+
+
+def rename_schema(schema_string, rename_func=lambda name: name):
+    """Create a RenamedSchema, where types and query root fields are renamed using rename_func.
+
+    Args:
+        schema_string: string describing a valid schema that does not contain extensions or
+                       input object definitions
+        rename_func: callable that takes string to string, used to transform the names of
```

Comment:
Not sure what value we get from setting a default value here. Isn't the identity renaming equivalent to just not calling this function at all?

Reply:
Calling it with the identity function would indeed not modify the schema. Nonetheless it would create a RenamedSchema with the right type and field maps, that allows the schema to be later merged with other schemas. 

I'm not totally sure if this design is good -- a function will take in a OrderedDict mapping schema name to RenamedSchema, and output a MergedSchema.

Reply:
Ah now I fully understand the conversation we had in person. I would not make the merging step be aware of the fact that it operates on renamed schemas. I think from its perspective, it should just be merging arbitrary schemas, and if they happen to be renamed, that would be handled by a separate step.

In other words, I think there's a primitive "merge schemas" function that is oblivious of the need for renaming. This function may then be used in a context where we know renaming happens. I think this approach is more easily understandable and testable.